tests/buffer/buffer_constructors_common.h

```diff
@@ -11,12 +11,60 @@
 
 #include "../common/common.h"
 
-namespace {
+namespace buffer_constructors_common {
 using namespace sycl_cts;
 
 template <typename T, int size, int dims>
 class BufferInteropNoEvent;
 
+template <typename T, int dims, typename allocT> class BufferCheck;
+
+template <int dims>
+bool check_range(cl::sycl::range<dims> r1, cl::sycl::range<dims> r2) {
+  bool result = true;
+  for (size_t i = 0; i < dims; ++i) {
+    result &= (r1[i] == r2[i]);
+  }
+  return result;
+}
+
+template <typename T, int dims, typename allocT>
+bool check_data(cl::sycl::buffer<T, dims, allocT> buf,
+                cl::sycl::range<dims> r) {
+  auto q = util::get_cts_object::queue();
+  int error = 0;
+  {
+    cl::sycl::buffer<int, 1> err_buf(&error, cl::sycl::range<1>(1));
+    q.submit([&](cl::sycl::handler &cgh) {
+
+      auto acc = buf.template get_access<cl::sycl::access::mode::read>(cgh);
+      auto err_acc =
+          err_buf.template get_access<cl::sycl::access::mode::read_write>(cgh);
+      cgh.parallel_for<BufferCheck<T, dims, allocT>>(
+          r, [=](cl::sycl::id<dims> idx) {
+            if (!check_equal_values(acc[idx], static_cast<T>(0))) {
```

Comment:
```suggestion
            if (!check_equal_values(acc[idx], T { 0 })) {
```
`T {}` would work too but probably more cryptic.